### ISSUE TITLE
Cannot get the companion app to work..

### ISSUE THREAD

**Issue by snivs** [LeoLombardi/tos-laimas-compass#4]
So, I even reinstalled full ToS from scratch, then cloned the repository, npm install, npm build and when I go npm start:

```
 npm build && npm start
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.12

> tos-laimas-compass@0.0.1 start C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass
> electron .


C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\settings.json: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\settings.json: Loaded
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\tracker.json: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\tracker.json: Not found
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior: Found
   MD5 698572c182bf5fb49b7942d2a68380b8 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\SumAni.ipf
   MD5 nil C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf
UPDATE: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf
   MD5 2c907933aa5f477cac4ac447812052ad C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\addonloader.lua
   MD5 2c907933aa5f477cac4ac447812052ad C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\addonloader.lua
SAME: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\addonloader.lua
   MD5 b2cffa61e0f1d26f074cd68d79ab7076 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\laimascompass.lua
   MD5 b2cffa61e0f1d26f074cd68d79ab7076 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\laimascompass\laimascompass.lua
SAME: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\laimascompass\laimascompass.lua
listening on *:9183
   fileCopy ERR
{ [Error: EBUSY: resource busy or locked, open 'C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf']
  errno: -4082,
  code: 'EBUSY',
  syscall: 'open',
  path: 'C:\\Program Files (x86)\\Steam\\steamapps\\common\\TreeOfSavior\\data\\SumAni.ipf' }
>>>>>C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass
>>>>>C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\index.html
a user connected
```

Most noticeable is that `tracker.json` is missing and `SumAni.ipf` is locked IToS is running, do I have to start the companion app before starting ToS? It didn't seem to work either way), if I start laima's compass without ToS running I get:

C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass>npm start

> tos-laimas-compass@0.0.1 start C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass
> electron .

```
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\settings.json: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\settings.json: Loaded
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\tracker.json: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\tracker.json: Not found
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior: Checking
C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior: Found
   MD5 698572c182bf5fb49b7942d2a68380b8 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\SumAni.ipf
   MD5 698572c182bf5fb49b7942d2a68380b8 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf
SAME: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf
   MD5 2c907933aa5f477cac4ac447812052ad C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\addonloader.lua
   MD5 2c907933aa5f477cac4ac447812052ad C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\addonloader.lua
SAME: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\addonloader.lua
   MD5 b2cffa61e0f1d26f074cd68d79ab7076 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\laimascompass.lua
   MD5 b2cffa61e0f1d26f074cd68d79ab7076 C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\laimascompass\laimascompass.lua
SAME: C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\addons\laimascompass\laimascompass.lua
listening on *:9183
>>>>>C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass
>>>>>C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\index.html
a user connected
[0x7FF99D3A7050] ANOMALY: meaningless REX prefix used
```

 Ingame, I see the Load Addons button, It says laima's compass is loaded and information is getting recorded in `data.txt`, but the button keeps reappearing when I switch maps, and the companion app is "just dead". It doesn't read anything (it reads no data, but shows no console errors).

I really love the idea of this App, and it'd be amazing if there could be an ingame version of it.

**Comment by LeoLombardi** [comment]
Hey @snivs , Thanks for your feedback!

@Questionguy663 reported on another issue thread that the problem may lie on restrictive permissions under `C:\Program Files (x86)`. I didn't had time yet to test it, but I wouldn't deny the possibility.

Now, regarding the issues you experienced - 

> npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
> npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.12

These two events are expected - chokidar is a NPM module that allows for directory change monitoring, and the binaries that support it changes according to OS.

> C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\tos-laimas-compass\assets\data\tracker.json: Not found

This is OK as well - tracker.json is the file where your character's data is stored.

> { [Error: EBUSY: resource busy or locked, open 'C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior\data\SumAni.ipf'

This means that ToS was running - the file was locked. It is expected, and shouldn't be an issue if you had previously installed Excrulon's mods; it's exactly the same file.

> It says laima's compass is loaded and information is getting recorded in data.txt

Good. This is the hardest part. It means that the LUA addon is up and running.

> but the button keeps reappearing when I switch maps

Now that's slightly odd. Mind taking a screenshot of your addon folder? I may need to replicate your configuration in order to test.

> and the companion app is "just dead". It doesn't read anything (it reads no data and shows no console errors).

This happens because the Steam repository detector may have failed. To fix this, click on the gear button on the top right, and use the browse function to select the game directory (not the addon - it'll resolve itself automatically. I Hope. =) )

> I really love the idea of this App, and it'd be amazing if there could be an ingame version of it.

Thank you! I like the idea of being able to use my tablet as a secondary info screen, so that's one of the reasons I chose to implement it as a separate service. But who knows - as other developers keep implementing new APIs, it may be possible to port the logic and use ingame-only resources. One can hope!

Once again, thanks for testing it out!

**Comment by snivs** [comment]
> This means that ToS was running - the file was locked. It is expected, and shouldn't be an issue if you had previously installed Excrulon's mods; it's exactly the same file.

I hadn't installed excrulon's :l or any other addon, I was trying to run laima's compass in a clean ToS install, I didn't even have an addons folder at first.

> Now that's slightly odd. Mind taking a screenshot of your addon folder? I may need to replicate your configuration in order to test.

As I mentioned above, it was just a fresh install... now it only has the addons folder and files created by laima's compass.

> This happens because the Steam repository detector may have failed. To fix this, click on the gear button on the top right, and use the browse function to select the game directory (not the addon - it'll resolve itself automatically. I Hope. =) )

I have selected the game path so many times... yet it doesn't work for some reason, I find it weird, like I'm missing some silly detail.

> Thank you! I like the idea of being able to use my tablet as a secondary info screen, so that's one of the reasons I chose to implement it as a separate service. But who knows - as other developers keep implementing new APIs, it may be possible to port the logic and use ingame-only resources. One can hope!

Who knows, maybe I'll be the one to port it... if I ever get some time to do so, lol.

As of Now, I'm reinstalling all of steam to C:/Steam to see if anything changes.. (that's why there's no screenshot of my addons folder :/ )

**Comment by snivs** [comment]
After reinstalling ToS to `C:\Steam\steamapps\common\TreeOfSavior` the compass works nicely.
A wild guess is that the compainion app failing to load data would have something to do with the ToS folder route previously containing a space in `C:\Program Files (x86)\Steam\steamapps\common\TreeOfSavior`.
